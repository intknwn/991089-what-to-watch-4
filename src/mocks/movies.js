export default [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    previewImg: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Fantasy`,
    year: 2018,
    rating: 6.6,
    score: 240,
    description: `In an effort to thwart Grindelwald's plans of raising pure-blood wizards to rule over all non-magical beings, Albus Dumbledore enlists his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world.`,
    director: `David Yates`,
    cast: [`Eddie Redmayne`, `Katherine Waterston`, `Dan Fogler`],
    runtime: 134,
  },
  {
    name: `Bohemian Rhapsody`,
    previewImg: `img/bohemian-rhapsody.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Biography`,
    year: 2018,
    rating: 8.0,
    score: 450,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury, leading up to their famous performance at Live Aid (1985).`,
    director: `Bryan Singer`,
    cast: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    runtime: 134,
  },
  {
    name: `We need to talk about Kevin`,
    previewImg: `img/we-need-to-talk-about-kevin.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Thriller`,
    year: 2011,
    rating: 7.5,
    score: 138,
    description: `Kevin's mother struggles to love her strange child, despite the increasingly dangerous things he says and does as he grows up. But Kevin is just getting started, and his final act will be beyond anything anyone imagined.`,
    director: `Lynne Ramsay`,
    cast: [`Tilda Swinton`, `John C. Reilly`, `Ezra Miller`],
    runtime: 112,
  },
  {
    name: `What We Do in the Shadows`,
    previewImg: `img/what-we-do-in-the-shadows.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Comedy`,
    year: 2019,
    rating: 8.5,
    score: 36894,
    description: `A look into the daily (or rather, nightly) lives of three vampires, who've lived together for over 100 years, on Staten Island.`,
    director: `Jemaine Clement`,
    cast: [`Kayvan Novak`, `Matt Berry`, `Natasia Demetriou`],
    runtime: 86,
  },
  {
    name: `Revenant`,
    previewImg: `img/revenant.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genre: `Adventure`,
    year: 2015,
    rating: 8.0,
    score: 710693,
    description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
    director: `Alejandro G. Iñárritu`,
    cast: [`Leonardo DiCaprio`, `Tom Hardy`, `Will Poulter`],
    runtime: 156,
  },
  {
    name: `Johnny English`,
    previewImg: `img/johnny-english.jpg`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Comedy`,
    year: 2003,
    rating: 6.2,
    score: 149199,
    description: `After a sudden attack on MI5, Johnny English, Britain's most confident, yet unintelligent spy, becomes Britain's only spy.`,
    director: `Peter Howitt`,
    cast: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    runtime: 89,
  },
  {
    name: `Falling Down`,
    previewImg: `https://via.placeholder.com/280x175`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Crime`,
    year: 1993,
    rating: 7.6,
    score: 173264,
    description: `An ordinary man frustrated with the various flaws he sees in society begins to psychotically and violently lash out against them.`,
    director: `Joel Schumacher`,
    cast: [`Michael Douglas`, `Robert Duvall`, `Barbara Hershey`],
    runtime: 113,
  },
  {
    name: `Batman & Robin`,
    previewImg: `https://via.placeholder.com/280x175`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Action`,
    year: 1997,
    rating: 3.8,
    score: 235550,
    description: `Batman and Robin try to keep their relationship together even as they must stop Mr. Freeze and Poison Ivy from freezing Gotham City.`,
    director: `Joel Schumacher`,
    cast: [`Arnold Schwarzenegger`, `George Clooney`, `Chris O'Donnell`],
    runtime: 125,
  },
  {
    name: `Terminator 2: Judgment Day`,
    previewImg: `https://via.placeholder.com/280x175`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Action`,
    year: 1991,
    rating: 8.5,
    score: 1002326,
    description: `A cyborg, identical to the one who failed to kill Sarah Connor, must now protect her ten year old son, John Connor, from a more advanced and powerful cyborg.`,
    director: `James Cameron`,
    cast: [`Arnold Schwarzenegger`, `Linda Hamilton`, `Edward Furlong`],
    runtime: 125,
  },
  {
    name: `Blade Runner 2049`,
    previewImg: `https://via.placeholder.com/280x175`,
    backgroundImg: `https://via.placeholder.com/1300x552`,
    posterImg: `https://via.placeholder.com/273x410`,
    previewVid: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    genre: `Action`,
    year: 2017,
    rating: 8.0,
    score: 469353,
    description: `Young Blade Runner K's discovery of a long-buried secret leads him to track down former Blade Runner Rick Deckard, who's been missing for thirty years.`,
    director: `Denis Villeneuve`,
    cast: [`Harrison Ford`, `Ryan Gosling`, `Ana de Armas`],
    runtime: 164,
  },
];
