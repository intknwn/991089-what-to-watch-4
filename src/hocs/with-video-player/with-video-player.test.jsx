import React from 'react';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import withVideoPlayer from './with-video-player.jsx';
import movies from '../../mocks/movies.js';

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node)
  ]).isRequired,
};

const MockComponentWrapped = withVideoPlayer(MockComponent);

it(`withVideoPlayer component should render correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      movie={movies[0]}
      playerConfig={{}}
      onExitClick={() => {}}
    >
      <React.Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
