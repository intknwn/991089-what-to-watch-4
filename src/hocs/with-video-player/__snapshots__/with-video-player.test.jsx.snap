// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withVideoPlayer component should render correctly 1`] = `
<div>
  <video
    className="player__video"
    onClick={[Function]}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
</div>
`;
